packages/lib/src/core/Analytics/AnalyticsInfoEvent.ts

```diff
@@ -0,0 +1,43 @@
+import { AnalyticsEvent } from './AnalyticsEvent';
+import { ANALYTICS_VALIDATION_ERROR_STR } from './constants';
+import { mapErrorCodesForAnalytics } from './utils';
+import { AnalyticsInfoEventObject } from './types';
+
+export class AnalyticsInfoEvent extends AnalyticsEvent {
+    public type: string;
+    public target: string;
+    public component: string;
```

Comment:
Fields that are defined on the parent object do not need to be re-defined on the extending classes. You can get rid of `component` field here and in the other events